prow/plank/plank.go

```diff
@@ -27,16 +27,18 @@ import (
 
 // NewProwJob initializes a ProwJob out of a ProwJobSpec.
 func NewProwJob(spec kube.ProwJobSpec) kube.ProwJob {
+	jobUUID := uuid.NewV1().String()
 	return kube.ProwJob{
 		APIVersion: "prow.k8s.io/v1",
 		Kind:       "ProwJob",
 		Metadata: kube.ObjectMeta{
-			Name: uuid.NewV1().String(),
+			Name: jobUUID,
 		},
 		Spec: spec,
 		Status: kube.ProwJobStatus{
 			StartTime: time.Now(),
 			State:     kube.TriggeredState,
+			UUID:      jobUUID,
```

Comment:
See above comment, I wanted to make it easy to avoid coupling in the future. The name field seems to be specifically used for naming prow pods which potentially has different requirements form the metadata bucket.